source-hubspot-native/test.flow.yaml

```diff
@@ -7,35 +7,121 @@ captures:
       local:
         command:
           - python
-          # - "-m"
-          # - "debugpy"
-          # - "--listen"
-          # - "0.0.0.0:5678"
-          # - "--wait-for-client"
           - "-m"
           - source_hubspot_native
         config: config.yaml
     bindings:
       - resource:
           name: companies
+          interval: PT420S
```

Comment:
What's the reason for adding this explicit interval to the test configuration?

Reply:
If i got it right, when a user does not generate their own bindings, we generate then by the default `common.Resource` resource, and the default wait value for estuary is 7 minutes. That's why this specific value at this place ( since discovery would see that no interval was set, i've added that ) .
Does that make sense?

Reply:
Yeah, that does make sense. I am wondering how/why these bindings were originally set to have no explicit `interval`. I don't 100% know what happens in that case, and have raised the question in the #saas-connectors slack channel.

And 7 minutes is a pretty weird value, lol. We use 30 seconds in a lot of other places, or maybe 5 minutes, but I'm not sure about a default wait value for estuary of 7 minutes?

Reply:
I dont know why but in my head 7 minutes was the default interval. Really don't remember where that came from ( probably need my meds ). @williamhbaker Should i leave the default interval of 30 seconds?

Reply:
30 seconds seems like a reasonable default interval for bindings that do not otherwise have restrictive rate limits, and are incremental in nature. We wouldn't want to re-fetch a huge snapshot every 30 seconds for example, so bindings like that should have larger intervals. I'll leave it to you to decide on the intervals based on your knowledge of how hubspot works, since I'm not very familiar with it other than these general statements.

Reply:
Another note is that I don't know why these pre-existing bindings had no interval at all, and were set to just re-fetch as fast as possible. Maybe there's a good reason for that?

Reply:
I think because hubspot lets you and we can get real-time updates this way.

Reply:
@dyaffe @williamhbaker  Hubspot limits us by 100 requests every 10 seconds, and given that we have 27+ streams, i think having a 30+ seconds breather for each stream would really help to not reach the limit. I will test with some values and get more data on this matter

Reply:
@williamhbaker Added intervals in 4171ca4368a0cd29e8af7b77533686cb6c80a249

Reply:
I'm still seeing this `PT420S` interval in `test.flow.yaml`, which I don't think is what we want.

Reply:
@williamhbaker Forgot to run discover, rebased the commit here da3558d73fb91d5fa30662cc2462b84f6fe451fb

Reply:
@williamhbaker Forgot to run discover, rebased the commit here fda33783a560759e3bdc40251a9633277d602c9f

Reply:
@Luishfs just one more push here.  We spoke about having 27 streams and some of them have 3 api calls per stream.  The worst case scenario seems to me that every single one has 3 calls.  If that's the case, we could lower the limit from 30s to 10s with room to spare since 10s would be at max 81 API requests every 10s.

Let me know if I'm wrong for any reason.